src/sentry/utils/auth.py

```diff
@@ -338,43 +348,45 @@ def log_auth_success(request, username, organization_id=None, source=None):
     )
 
 
-def log_auth_failure(request, username=None):
+def log_auth_failure(request: HttpRequest, username: str | None = None) -> None:
     logger.info(
         "user.auth.fail", extra={"ip_address": request.META["REMOTE_ADDR"], "username": username}
     )
 
 
-def has_user_registration():
+def has_user_registration() -> bool:
     from sentry import features, options
 
     return features.has("auth:register") and options.get("auth.allow-registration")
 
 
-def is_user_signed_request(request):
+def is_user_signed_request(request: HttpRequest) -> bool:
     """
     This function returns True if the request is a signed valid link
     """
     try:
-        return request.user_from_signed_request
+        return bool(request.user_from_signed_request)
     except AttributeError:
         return False
 
 
-def set_active_org(request: Request, org_slug: str) -> None:
+def set_active_org(request: HttpRequest, org_slug: str) -> None:
```

Comment:
Really silly question: what is the different between `Request` and `HttpRequest`?

Reply:
The best answer I can give is that Django uses `HttpRequest` to model general incoming traffic, and `Request` is part of the `rest_framework` module and is more specific to DRF.

I only know that `HttpRequest` is correct here by live-testing it. I don't have a safe heuristic to tell when to use which. There's a chance we may even need `HttpRequest | Request` somewhere. 🙁